### PR TITLE
Fix Scraping Error and other QoL Improvements

### PR DESCRIPTION
**Pull request by tim-hy**
Hi! This PR fixes the broken web-scraping and adds a few other QoL improvements.

**TL;DR**
1. Fixed web scraping
2. Added new country column
3. Updated Winter 2022 transfers
4. Added new league (but not its data, yet), with potential for more (examples [here](https://github.com/tim-hy/tmarkt-transfers/tree/dev/data))

---

**1. Fix broken web-scrape**

It appears transfermarkt may have restructured their website, meaning that the class names in the transfers pages have been renamed. I am unsure what it looked like previously, but the html lookup has now been edited in `src/functions.R` [here](https://github.com/tim-hy/tmarkt-transfers/blob/25bdef96396b5c8aa2af71e1e9573faebe75bfd9/src/functions.R#L29) to retrieve the correct data: 

```diff
>>> git diff master:src/functions.R bugfix:src/functions.R
- html_elements(".table-header") %>%
- html_text2() 
+ html_elements(".content-box-headline--logo") %>%
+ html_text2() 
```

I was also having lots of issues with handshake errors/timeouts. To prevent, this, have increased retrieval timeout using the `httr` package, and sleep time is increased to 5 seconds [here](https://github.com/tim-hy/tmarkt-transfers/blob/25bdef96396b5c8aa2af71e1e9573faebe75bfd9/src/functions.R#L24-L25):

```diff
>>> git diff master:src/functions.R bugfix:src/functions.R
-  page <- read_html(transfers_url)
+  cat("Scraping", transfers_url, "\n")
+  page <-  GET(transfers_url, config = httr::config(connecttimeout = 120)) %>% read_html
```

Returning a `NULL` value when no transfers are found has been replaced with an empty dataframe, to prevent errors when merging.

**2. Country column**

An additional `country` column has been added the the input & output data. This provides useful context (especially when multiple leagues are merged together into a single dataframe, as names can become confusing) [here](https://github.com/tim-hy/tmarkt-transfers/blob/25bdef96396b5c8aa2af71e1e9573faebe75bfd9/src/scrape-history.R#L20-L24):

```diff
>>>git diff master:src/scrape-history.R bugfix:src/scrape-history.R 
-transfers <- map2(
-  league_meta$league_name, league_meta$league_id,
+transfers <- pmap(
+  list(league_meta$league_name, league_meta$league_id, league_meta$country),
   ~ map_dfr(seasons, get_transfers_history,
-            league_name = .x, league_id = .y)
+            league_name = ..1, league_id = ..2, country=..3)
   )
```

**3. Winter 2022 Transfer update**

As the Automated run failed, the Winter 2022 transfers have been updated (along with the additional "country" column.)

**4. Expanded config**

Expanded config file with additional league (Juliper Pro League). I refrained from adding lots of new leagues, and the Juliper Pro League output data has not been added to this PR as I don't if it was wanted. If desired, I do have data from multiple other leagues in my own fork here: https://github.com/tim-hy/tmarkt-transfers . 

Let me know what you think of all this and if you have any suggestions/changes. Thanks!

### CODE REVIEW

**Comment by tim-hy** [comment]
Just discovered that the Winter 2022 updates https://github.com/ewenme/transfers/pull/11/commits/216b6d9e1838bedbb6d39a12844acd5c4f65106f have lower case window names (`summer`/`winter` instead of `Summer`/`Winter`) and refer to the `league_id` instead of the `league_name` (for instance `Gb1` instead of `Premier League`). Apologies, will fix.

---
**[EDIT]**
Just found the issue with the capitalisation differences. At 92755b3, these lines are added https://github.com/ewenme/transfers/blob/92755b3d124cae316cf3aec2d479de26edf5a56d/src/functions.R#L6-L9

with summer/winter in lower case, whilst the previous version used upper case (old lines 104-108 in `functions.R` at https://github.com/ewenme/transfers/commit/92755b3d124cae316cf3aec2d479de26edf5a56d#). I will update functions to use upper case, to match the previous files.

**Comment by ewenme** [comment]
hey @tim-hy looks wicked. I've removed the expanded config file from the PR but everything else works for me. Ta!
